btp-document/Smart-Contracts-PRA.md

```diff
@@ -0,0 +1,530 @@
+```
+
+You will receive information similar as:
+
+```bash
+$ truffle(Moonriverlocal)> bmv.address
+'0x7acc1aC65892CF3547b1b0590066FB93199b430D'
+```
+
+**Note that:** Please do not skip these steps. This information will later be used to register ***Veriffier*** (`addVerifier()`) in ***BMCManagement*** contract
+
+### 4. Config Moonriver BMC
+
+____
+
+**Note that: This configuration step must be executed after completely deploying smart contracts (BSH, BMC, and BMV) on both connecting networks. In this example, we would like to show how to config to setup a connection between ICON and Moonriver networks. For other cases, there might be a different configuration**
+
+- Launch Truffle console
+
+```bash
+$ cd ${BTP_PROJ_DIR}/solidity/bmc
+
+$ truffle console --network Moonriverlocal
+```
+
+- Add **BMV**
+
+```bash
+$ truffle(Moonriverlocal)> let bmcManagement = await BMCManagement.deployed()
+
+$ truffle(Moonriverlocal)> let bmcPeriphery = await BMCPeriphery.deployed()
+
+$ truffle(Moonriverlocal)> await bmcManagement.addVerifier("0x3.icon", bmv-contract-address)
+# For example: bmv-contract-address = "0x7acc1aC65892CF3547b1b0590066FB93199b430D"
+```
+
+- Add a connection link to ICON's BMC
+
+```bash
+$ truffle(Moonriverlocal)> await bmcManagement.addLink("btp://0x3.icon/icon-bmc-contract-address")
+# For example: icon-bmc-contract-address = cx8eb24849a7ceb16b8fa537f5a8b378c6af4a0247
+```
+
+The above command helps to set a connection link from Moonriver-BMC to ICON-BMC with default setting values:
+
+&emsp; &emsp; &emsp; + `BLOCK_INTERVAL_MSEC`: block interval of ICON (default = 1000ms)
+
+&emsp; &emsp; &emsp; + `MAX_AGGREGATION`: max_aggregation value (default = 10)
+
+&emsp; &emsp; &emsp; + `DELAY_LIMIT`: acceptance of delayed submission sending from BMR to BMC (default = 3)
+
+- Set Link Configuration: this step will help you to change default setting values from the command above
+
+```bash
+$ truffle(Moonriverlocal)> await bmcManagement.setLink("btp://0x3.icon/icon-bmc-contract-address", 3000, 5, 3)
+# For example: icon-bmc-contract-address = cx8eb24849a7ceb16b8fa537f5a8b378c6af4a0247
+```
+
+- Add BSH Service contract
+
+```bash
+$ truffle(Moonriverlocal)> await bmcManagement.addService("CoinTransfer", bsh-contract-address)
+# For example: bsh-contract-address = "0xD5Cd1ea4E9874FfB7d553c517Cec812B3069e322"
+```
+
+- Get Link Status: please **DO NOT** skip this step. This output will then be used to set parameters to deploy BMR
+
+```bash
+$ truffle(Moonriverlocal)> await bmcPeriphery.getStatus("btp://0x3.icon/icon-bmc-contract-address")
+# For example: icon-bmc-contract-address = cx8eb24849a7ceb16b8fa537f5a8b378c6af4a0247
+```
+
+### 5. Config Moonriver BSH
+
+____
+
+- Launch Truffle console
+
+```bash
+$ cd ${BTP_PROJ_DIR}/solidity/bmc
+
+$ truffle console --network Moonriverlocal
+```
+
+- Register 'ICX' token
+
+```bash
+$ truffle(Moonriverlocal)> let bshPeriphery = await BSHPeriphery.deployed()
```

Comment:
In order to load BSHPeriphery contract, you need to copy the contract definition .json files from `PRA-Contracts/btp/solidity/bsh/build/contracts/*` to `PRA-Contracts/btp/solidity/bmc` before running truffle console

Reply:
That's a typo. It suppose to be executed under the directory `solidity/bsh` instead.